File: android/app/src/main/java/com/example/app/ui/conversation/ConversationInputBar.kt
package caddypro.ui.conversation

import androidx.compose.animation.AnimatedContent
import androidx.compose.animation.fadeIn
import androidx.compose.animation.fadeOut
import androidx.compose.animation.togetherWith
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.text.KeyboardActions
import androidx.compose.foundation.text.KeyboardOptions
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.Send
import androidx.compose.material.icons.filled.Mic
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.IconButtonDefaults
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.OutlinedTextField
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.input.ImeAction
import androidx.compose.ui.text.input.KeyboardCapitalization
import androidx.compose.ui.unit.dp

/**
 * Input bar for the conversation screen.
 *
 * Features:
 * - TextField for text input
 * - Mic button (toggles to Send when text present)
 * - Animated state transitions
 * - Keyboard handling
 *
 * Spec reference: navcaddy-engine.md R1, R7
 * Plan reference: navcaddy-engine-plan.md Task 18
 */
@Composable
fun ConversationInputBar(
    value: String,
    onValueChange: (String) -> Unit,
    onSend: () -> Unit,
    onMicClick: () -> Unit,
    isVoiceActive: Boolean,
    enabled: Boolean,
    modifier: Modifier = Modifier
) {
    Surface(
        modifier = modifier.fillMaxWidth(),
        color = MaterialTheme.colorScheme.surface,
        shadowElevation = 8.dp
    ) {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(horizontal = 16.dp, vertical = 8.dp),
            verticalAlignment = Alignment.Bottom
        ) {
            // Text input field
            OutlinedTextField(
                value = value,
                onValueChange = onValueChange,
                modifier = Modifier
                    .weight(1f)
                    .padding(end = 8.dp),
                placeholder = {
                    Text(
                        text = if (isVoiceActive) "Listening..." else "Ask Bones anything...",
                        style = MaterialTheme.typography.bodyLarge
                    )
                },
                enabled = enabled && !isVoiceActive,
                keyboardOptions = KeyboardOptions(
                    capitalization = KeyboardCapitalization.Sentences,
                    imeAction = ImeAction.Send
                ),
                keyboardActions = KeyboardActions(
                    onSend = {
                        if (value.isNotBlank()) {
                            onSend()
                        }
                    }
                ),
                maxLines = 4,
                textStyle = MaterialTheme.typography.bodyLarge,
                shape = MaterialTheme.shapes.large
            )

            // Mic/Send button with animated transition
            AnimatedContent(
                targetState = value.isNotBlank(),
                transitionSpec = {
                    fadeIn() togetherWith fadeOut()
                },
                label = "Input button animation"
            ) { hasText ->
                if (hasText) {
                    // Send button
                    IconButton(
                        onClick = onSend,
                        enabled = enabled,
                        colors = IconButtonDefaults.iconButtonColors(
                            containerColor = MaterialTheme.colorScheme.primary,
                            contentColor = MaterialTheme.colorScheme.onPrimary
                        ),
                        modifier = Modifier.size(48.dp)
                    ) {
                        Icon(
                            imageVector = Icons.AutoMirrored.Filled.Send,
                            contentDescription = "Send message"
                        )
                    }
                } else {
                    // Mic button
                    IconButton(
                        onClick = onMicClick,
                        enabled = enabled,
                        colors = IconButtonDefaults.iconButtonColors(
                            containerColor = if (isVoiceActive) {
                                MaterialTheme.colorScheme.error
                            } else {
                                MaterialTheme.colorScheme.secondaryContainer
                            },
                            contentColor = if (isVoiceActive) {
                                MaterialTheme.colorScheme.onError
                            } else {
                                MaterialTheme.colorScheme.onSecondaryContainer
                            }
                        ),
                        modifier = Modifier.size(48.dp)
                    ) {
                        Icon(
                            imageVector = Icons.Default.Mic,
                            contentDescription = if (isVoiceActive) {
                                "Stop voice input"
                            } else {
                                "Start voice input"
                            }
                        )
                    }
                }
            }
        }
    }
}

File: ios/App/Sources/UI/Conversation/ConversationInputBar.swift
import SwiftUI

/**
 * Input bar for the conversation screen.
 *
 * Features:
 * - TextField for text input
 * - Mic button (toggles to Send when text present)
 * - Animated state transitions
 * - Keyboard handling
 *
 * Spec reference: navcaddy-engine.md R1, R7
 * Plan reference: navcaddy-engine-plan.md Task 19
 */
struct ConversationInputBar: View {
    let value: String
    let onValueChange: (String) -> Void
    let onSend: () -> Void
    let onMicClick: () -> Void
    let isVoiceActive: Bool
    let enabled: Bool

    var body: some View {
        VStack(spacing: 0) {
            Divider()

            HStack(alignment: .bottom, spacing: 8) {
                // Text input field
                TextField(
                    isVoiceActive ? "Listening..." : "Ask Bones anything...",
                    text: Binding(
                        get: { value },
                        set: { onValueChange($0) }
                    ),
                    axis: .vertical
                )
                .textFieldStyle(.roundedBorder)
                .lineLimit(1...4)
                .disabled(!enabled || isVoiceActive)
                .submitLabel(.send)
                .onSubmit {
                    if !value.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty {
                        onSend()
                    }
                }
                .accessibilityLabel("Message input")
                .accessibilityHint("Enter your question for Bones")

                // Mic/Send button with animated transition
                ZStack {
                    if !value.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty {
                        // Send button
                        Button(action: onSend) {
                            Image(systemName: "arrow.up.circle.fill")
                                .font(.system(size: 32))
                                .foregroundStyle(.white)
                                .background(
                                    Circle()
                                        .fill(Color.accentColor)
                                        .frame(width: 32, height: 32)
                                )
                        }
                        .disabled(!enabled)
                        .accessibilityLabel("Send message")
                        .transition(.scale.combined(with: .opacity))
                    } else {
                        // Mic button
                        Button(action: onMicClick) {
                            Image(systemName: "mic.fill")
                                .font(.system(size: 20))
                                .foregroundStyle(isVoiceActive ? .white : .primary)
                                .frame(width: 44, height: 44)
                                .background(
                                    Circle()
                                        .fill(isVoiceActive ? Color.red : Color(.tertiarySystemBackground))
                                )
                        }
                        .disabled(!enabled)
                        .accessibilityLabel(isVoiceActive ? "Stop voice input" : "Start voice input")
                        .transition(.scale.combined(with: .opacity))
                    }
                }
                .frame(width: 44, height: 44)
                .animation(.spring(response: 0.3, dampingFraction: 0.7), value: value.isEmpty)
            }
            .padding(.horizontal, 16)
            .padding(.vertical, 8)
        }
        .background(Color(.systemBackground))
        .shadow(color: .black.opacity(0.05), radius: 8, y: -2)
    }
}

// MARK: - Previews

#Preview("Empty Input") {
    ConversationInputBar(
        value: "",
        onValueChange: { _ in },
        onSend: {},
        onMicClick: {},
        isVoiceActive: false,
        enabled: true
    )
    .previewLayout(.sizeThatFits)
}

#Preview("With Text") {
    ConversationInputBar(
        value: "What club should I use?",
        onValueChange: { _ in },
        onSend: {},
        onMicClick: {},
        isVoiceActive: false,
        enabled: true
    )
    .previewLayout(.sizeThatFits)
}

#Preview("Voice Active") {
    ConversationInputBar(
        value: "",
        onValueChange: { _ in },
        onSend: {},
        onMicClick: {},
        isVoiceActive: true,
        enabled: true
    )
    .previewLayout(.sizeThatFits)
}

#Preview("Disabled") {
    ConversationInputBar(
        value: "",
        onValueChange: { _ in },
        onSend: {},
        onMicClick: {},
        isVoiceActive: false,
        enabled: false
    )
    .previewLayout(.sizeThatFits)
}
